tools/geom.go

```diff
@@ -0,0 +1,450 @@
+
+	xyPairs, err := getDataPairsfromTextBlock("XS GIS Cut Line", sc, 64, 16)
+	if err != nil {
+		return layer, xyPairs, 0.0, err
+	}
+
+	mzPairs, err := getDataPairsfromTextBlock("#Sta/Elev", sc, 80, 8)
+	if err != nil {
+		return layer, xyPairs, mzPairs[0][0], err
+	}
+
+	xyzPoints := attributeZ(xyPairs, mzPairs)
+
+	xyzLineString := gdal.Create(gdal.GT_LineString25D)
+	for _, point := range xyzPoints {
+		xyzLineString.AddPoint(point.x, point.y, point.z)
+	}
+
+	xyzLineString.Transform(transform)
+	// This is a temporary fix since the x and y values need to be flipped
+	yxzLineString := flipXYLineString25D(xyzLineString)
+
+	multiLineString := yxzLineString.ForceToMultiLineString()
+	wkb, err := multiLineString.ToWKB()
+	if err != nil {
+		return layer, xyPairs, mzPairs[0][0], err
+	}
+	layer.Geometry = wkb
+	return layer, xyPairs, mzPairs[0][0], err
+}
+
+func getBanks(line string, transform gdal.CoordinateTransform, xsLayer vectorLayer, xyPairs [][2]float64, startingStation float64) ([]vectorLayer, error) {
+	layers := []vectorLayer{}
+
+	bankStations := strings.Split(rightofEquals(line), ",")
+	for _, s := range bankStations {
+		layer := vectorLayer{FeatureName: strings.TrimSpace(s), Fields: map[string]interface{}{}}
+		layer.Fields["RiverReachName"] = xsLayer.Fields["RiverReachName"]
+		layer.Fields["xsName"] = xsLayer.FeatureName
+		bankStation, err := strconv.ParseFloat(s, 64)
+		if err != nil {
+			return layers, err
+		}
+		bankXY := interpXY(xyPairs, bankStation-startingStation)
+		xyPoint := gdal.Create(gdal.GT_Point)
+		xyPoint.AddPoint2D(bankXY[0], bankXY[1])
+		xyPoint.Transform(transform)
+		// This is a temporary fix since the x and y values need to be flipped
+		yxPoint := flipXYPoint(xyPoint)
+		multiPoint := yxPoint.ForceToMultiPoint()
+		wkb, err := multiPoint.ToWKB()
+		if err != nil {
+			return layers, err
+		}
+		layer.Geometry = wkb
+		layers = append(layers, layer)
+	}
+	return layers, nil
+}
+
+// GetGeospatialData ...
+func GetGeospatialData(gd *GeoData, fs filestore.FileStore, geomFilePath string) error {
+	geomFileName := filepath.Base(geomFilePath)
+	f := Features{}
+	riverReachName := ""
+
+	file, err := fs.GetObject(geomFilePath)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	sc := bufio.NewScanner(file)
+
+	transform, err := getTransform(sourceCRS, DestinationCRS)
+	if err != nil {
+		return err
+	}
+
+	for sc.Scan() {
+		line := sc.Text()
+		switch {
+		case strings.HasPrefix(line, "River Reach="):
+			riverLayer, err := getRiverCenterline(sc, transform)
+			if err != nil {
+				return err
+			}
+			f.Rivers = append(f.Rivers, riverLayer)
+			riverReachName = riverLayer.FeatureName
+
+		case strings.HasPrefix(line, "Type RM Length L Ch R = 1"):
+			xsLayer, bankLayers, err := getXSBanks(sc, transform, riverReachName)
+			if err != nil {
+				return err
+			}
+			f.XS = append(f.XS, xsLayer)
+			f.Banks = append(f.Banks, bankLayers...)
+		}
+	}
+
+	gd.Features[geomFileName] = f
+	return nil
+}
```

Comment:
I would consider separating this into a couple files, especially if you think it will grow later on. I could see one for geom generics, one for banks, one for cross sections, etc.

Reply:
Good idea, I just added issue #5 and will break these apart in the other branch which is ahead.